Library/Homebrew/cmd/update.rb

```diff
@@ -92,10 +95,25 @@ def update
       end
     end
 
-    if report.empty?
+    # Report changes to Homebrew itself and tapped Git repositories.
+    repo_changes.reject! { |_key, r| r[0] == r[1] }
```

Comment:
It would be faster to check for this *before* adding a change in the hash instead of adding everything then removing unneeded stuff.